src/Version.php

```diff
@@ -64,7 +64,7 @@ public function __get(string $property)
      */
     public function __toString(): string
     {
-        $version = implode('.', [$this->major, $this->minor, $this->patch]);
+        $version = "{$this->major}.{$this->minor}.{$this->patch}";
```

Comment:
better performance

Reply:
Again, performance is negligible, even over thousands of iterations.

```
> timeit -n 1000 implode('.', ['foo', 'bar', 'baz']);
= "foo.bar.baz"

Command took 0.000001 seconds on average (0.000001 median; 0.000759 total) to complete.

> timeit -n 1000 'foo' . '.' . 'bar' . '.' . 'baz'
= "foo.bar.baz"

Command took 0.000000 seconds on average (0.000000 median; 0.000475 total) to complete.
```

```suggestion
        $version = implode('.', [$this->major, $this->minor, $this->patch]);
```

Reply:
You can decide. 🙃

Reply:
As I stated in my comment, let's strip this PR back to just the method + tests changes.

> If we can get this PR focused on only the necessary changes (i.e. `isPreRelease()` and `hasBuild()` methods + tests) we can get this merged.